src/lib/store/settings.ts

```diff
@@ -0,0 +1,41 @@
+import { defineStore } from 'pinia';
+
+export interface Settings {
+  websocketAddress: string;
+}
+
+function defaultSettings(): Settings {
+  return {
+    websocketAddress: 'ws://192.168.0.145:9090',
+  };
+}
+
+function getStoredSettings(): Settings {
+  const settings = localStorage.getItem('settings');
+  if (settings) {
+    return JSON.parse(settings);
+  }
+
+  return defaultSettings();
+}
+
+export const useSettingsStore = defineStore('settings', {
```

Comment:
I'll have to take a closer look, but I believe it's so that I can hook into local storage more easily. Also, the settings store is really just storing data, so Option seemed like a simpler approach. I'll try setup and see if it works better.

Reply:
yeah, I agree that the options api is usually cleaner and more explicit for stores in particular, but because we are using setup in our vue components, it's probably best if we try to have coded the same way for readable. Kind of like in react mixing in class and functional components together